kernel/lib/log.c

```diff
@@ -0,0 +1,41 @@
+//! @file log.c
+//! @brief File containing definitions for logging functions
+
+#include <lib/fmt.h>   // For vsnprintf
+#include <sys/ports.h> // For outb
+
+//! @brief Maximum length of the buffer for log_printf
+#define LOG_BUFFER_SIZE 4096
+
+//! @brief Print one character to kernel log
+//! @param character Character to be printed
+void log_putc(char character) {
+	// Print one character to QEMU debugcon
```

Comment:
Maybe put side comments like so
```c
outb(0xe9, (uint8_t)character); // Print one character to the debugging console
```

Reply:
This way people read the code first, and then the explanation. I think its more natural if you first read the objective, and then read the actual impl